### PR TITLE
Publishing nakadi.event.type.log as Avro

### PR DESCRIPTION
**Pull request by thomasabraham**
# Publishing nakadi.event.type.log as Avro

> Zalando ticket : team-aruha/issues/984

## Description
Publishing the `nakadi.event.type.log` events as Avro.

### CODE REVIEW

File: acceptance-test/src/acceptance-test/java/org/zalando/nakadi/webservice/BinaryEventPublisherAT.java [hunk 1]
```diff
@@ -60,15 +68,56 @@ public void testNakadiSubscriptionLogInAvro() throws Exception {
...
+        Assert.assertEquals(
+                NAKADI_BATCH_PUBLISHED,
+                ((Map) event.get("metadata")).get("event_type"));
+    }
+
+    private Map consumeEvent(final TestStreamingClient client) {
+        TestUtils.waitFor(() -> MatcherAssert.assertThat(
+                client.getBatches().size(), Matchers.greaterThanOrEqualTo(1)), 10000);
+        final List<Map> events = client.getBatches().get(0).getEvents();
+        Assert.assertFalse(events.isEmpty());
```
[the remark this answers is not in the document]
Comment (thomasabraham) [reply acceptance-test/src/acceptance-test/java/org/zalando/nakadi/webservice/BinaryEventPublisherAT.java:120]: Good point, I have already re-worked this in the followup PR, so I'll include this there.

File: core-common/src/main/java/org/zalando/nakadi/domain/kpi/KPIEvent.java [hunk 2]
```diff
@@ -12,10 +12,13 @@ public abstract class KPIEvent {
...
-        which is why this method is named "eventTypeOfThisKPIEvent()
-     */
-    public abstract String eventTypeOfThisKPIEvent();
+    private final String eventTypeOfThisKPIEvent;
+
+    protected KPIEvent(final String eventTypeOfThisKPIEvent) {
+        this.eventTypeOfThisKPIEvent = eventTypeOfThisKPIEvent;
+    }
+
+    public final String eventTypeOfThisKPIEvent() {
```
[the remark this answers is not in the document]
Comment (thomasabraham) [reply core-common/src/main/java/org/zalando/nakadi/domain/kpi/KPIEvent.java:21]: I think we can; originally I put a complicated name to minimize the chance of this getting confused with the getter of an actual field of the KPI event-type. 
I think so far I haven't found a KPI event-type with a `name` field, so this should be fine. 
I'll do this.

File: acceptance-test/src/acceptance-test/java/org/zalando/nakadi/webservice/BinaryEventPublisherAT.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (thomasabraham) [reply acceptance-test/src/acceptance-test/java/org/zalando/nakadi/webservice/BinaryEventPublisherAT.java]: This is done

File: core-common/src/main/java/org/zalando/nakadi/domain/kpi/KPIEvent.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (thomasabraham) [reply core-common/src/main/java/org/zalando/nakadi/domain/kpi/KPIEvent.java]: This is done

File: core-common/src/main/java/org/zalando/nakadi/domain/kpi/EventTypeLogEvent.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    @KPIField("event_type")
+    private String eventType;
+    @KPIField("status")
+    private String status;
+    @KPIField("category")
+    private String category;
+    @KPIField("authz")
+    private String authz;
+    @KPIField("compatibility_mode")
+    private String compatibilityMode;
```
[the remark this answers is not in the document]
Comment (thomasabraham) [reply core-common/src/main/java/org/zalando/nakadi/domain/kpi/EventTypeLogEvent.java:15]: I'm calling `name()` method on the enum while setting it to the String value to keep it simple. 
I would try to use enum as such in a subsequent iteration.

**Comment by adyach** [comment]
👍

**Comment by thomasabraham** [comment]
:+1:

**Pull request merged**